lib/services/MetricService.ts

```diff
@@ -0,0 +1,206 @@
+   * @property {Registry} requestDuration   - The default metrics registry (if enabled)
+   */
+  private registries: { core: Registry, default?: Registry, requestDuration?: Registry };
+
+  /**
+   * @param {PrometheusPluginConfiguration} config - The plugin configuration
+   */
+  constructor(config: PrometheusPluginConfiguration) {
+    this.registries = {
+      core: new Registry(),
+    };
+
+    this.metrics = {
+      core: {
+        api: {
+          concurrentRequests: new Gauge({ 
+            name: `${config.metrics.core.prefix}api_concurrent_requests`, 
+            help: 'Number of concurrent requests', 
+            registers: [this.registries.core] 
+          }),
+          pendingRequests: new Gauge({ 
+            name: `${config.metrics.core.prefix}api_pending_requests`, 
+            help: 'Number of pending requests', 
+            registers: [this.registries.core] 
+          }),
+        },
+        network: {
+          connections: new Gauge({ 
+            name: `${config.metrics.core.prefix}network_connections`,
+            help: 'Number of connections', 
+            labelNames: ['protocol'], 
+            registers: [this.registries.core] 
+          }),
+        },
+        realtime: {
+          rooms: new Gauge({
+            name: `${config.metrics.core.prefix}realtime_rooms`, 
+            help: 'Number of rooms', 
+            registers: [this.registries.core] 
+          }),
+          subscriptions: new Gauge({
+            name: `${config.metrics.core.prefix}realtime_subscriptions`,
+            help: 'Number of subscriptions', 
+            registers: [this.registries.core] 
+          }),
+        },
+      },
+    };
+
+    if (config.metrics.core.monitorRequestDuration) {
+      // Need to be registered in a separate registry to avoid
+      // the core registry resetting in updateCoreMetrics
+      this.registries.requestDuration = new Registry();
+      this.metrics.requestDuration = new Histogram({
+        name: `${config.metrics.core.prefix}api_request_duration_ms`,
+        help: 'Duration of Kuzzle requests in ms',
+        labelNames: ['action', 'controller', 'protocol', 'status'],
+        registers: [this.registries.requestDuration],
+        buckets: [0.10, 5, 15, 50, 100, 200, 300, 400, 500]
+      });
+    }
+
+    if (config.metrics.default.enabled) {
+      this.registries.default = new Registry();
+      collectDefaultMetrics({
+        register: this.registries.default,
+        prefix: config.metrics.default.prefix,
+        eventLoopMonitoringPrecision: config.metrics.default.eventLoopMonitoringPrecision,
+        gcDurationBuckets: config.metrics.default.gcDurationBuckets,
+      });
+    }
+  }
+
+  /**
+   * Update the Prometheus coreMetrics with from the server:metrics JSON response
+   * @param {JSONObject} jsonMetrics - The server:metrics JSON response
+   */
+  public updateCoreMetrics(jsonMetrics: JSONObject): void {
```

Comment:
Nitpicking: coding standard
```suggestion
  public updateCoreMetrics (jsonMetrics: JSONObject): void {
```